core/tests/Unit/FundingSource/Eligibility/Checker/FundingSourceEligibilityServiceTest.php

```diff
@@ -92,7 +92,32 @@ public function testEmptyEligibleFundingSources()
         $this->context->method('getShop')->willReturn($shop);
         $this->fundingSourcePresenter->method('getAllActiveForSpecificShop')->with(1)->willReturn([]);
 
-        self::assertEmpty($this->fundingSourceEligibilityService->getEligibleFundingSource());
+        self::assertEmpty($this->fundingSourceEligibilityService->getEligibleFundingSources());
+    }
+
+    public function testDisabledEligibleFundingSources()
+    {
+        $shop = $this->createMock(\Shop::class);
+        $shop->id = 1;
+        $this->context->method('getShop')->willReturn($shop);
+        $this->fundingSourcePresenter->method('getAllActiveForSpecificShop')->with(1)->willReturn([
+            new FundingSource('paypal', 'PayPal', 0, false, null),
+            new FundingSource('apple_pay', 'Apple Pay', 0, false, null),
+        ]);
+
+        self::assertEmpty($this->fundingSourceEligibilityService->getEligibleFundingSources());
+    }
+
+    public function testUnknownEligibilityChecker()
+    {
+        $shop = $this->createMock(\Shop::class);
+        $shop->id = 1;
+        $this->context->method('getShop')->willReturn($shop);
+        $this->fundingSourcePresenter->method('getAllActiveForSpecificShop')->with(1)->willReturn([
+            new FundingSource('paypal', 'PayPal', 0, true, null),
+        ]);
+
+        self::assertArrayHasKey('paypal', $this->fundingSourceEligibilityService->getEligibleFundingSources());
```

Comment:
To make sure that a funding source without a checker implementation is considered eligible.